### DISCUSSION TITLE
Is it possible to skip the current Router?

### DISCUSSION THREAD

**Discussion by Frog-kt** [honojs/hono#2608] [Q&A] [answered]
We are considering implementing a conditional branch within Middleware that would call the next router if there is no match.

Is there a way to implement the process of skipping the current router within Middleware?

Here is the ideal code I have in mind right now.

```ts
const checkVersion = (version: string) => {
	return createMiddleware(async (c, next) => {
		const apiVersion = c.req.header('api-version')
		if (apiVersion !== version) {
			return await next()
		}
		return await next()
	})
}

app.get('/api', checkVersion('1.0.0'), (c) => {
	return c.json({ version: 'world 1.0.0' })
})

app.get('/api', checkVersion('1.0.1'), (c) => {
	return c.json({ version: 'world 1.0.1' })
})

app.get('/api', (c) => {
	return c.json({ version: 'other' })
})
```

**Comment by yusukebe** [comment, marked as answer]
Hi @Frog-kt 

The answer may be no. There is no feature to skip the next handler.

**Reply by Frog-kt** [reply]
@yusukebe 
Thanks for the response.

Do you think this feature is practical from your point of view?
I'am wondering if I should create this feature and submit a PR.

**Reply by yusukebe** [reply]
If it can be implemented with less code and there is no performance degradation, we will consider it.
